opta/commands/deploy.py

```diff
@@ -156,3 +160,17 @@ def deploy(
         image_digest=image_digest,
         detailed_plan=detailed_plan,
     )
+
+
+def __check_layer_and_image(layer: "Layer", image: str) -> str:
+    k8s_module = layer.get_module_by_type("k8s-service")
+    image_name = k8s_module[0].data.get("image")
+    if image_name == "AUTO" and image is None:
+        raise UserErrors(
+            "An image should be passed when using `image` as AUTO in configuration"
+        )
+    if image_name != "AUTO" and image is not None:
+        raise UserErrors(
+            f"Do not pass any image. Image {image_name} already present in configuration."
```

Comment:
you can also add ` or set the image value to AUTO`

Reply:
Sure, will do.